File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def __init__(self):
        self.result = 0
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        def counter(noder, stringer):
            copy = stringer + str(noder.val)
            if noder.left:
                counter(noder.left,copy)
            if noder.right:
                counter(noder.right,copy)
            if noder.left == None and noder.right== None:
                self.result += int(copy)
        counter(root,"")
        return self.result

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int answer = 0;
    public int sumNumbers(TreeNode root) {
        String sr = "";
        searcher(root, sr);
        return answer;
    }
    public void searcher(TreeNode noder, String current){
        current += noder.val;
        if(noder.left == null && noder.right == null){
            answer += Integer.valueOf(current);
        }
        else{
            if(noder.left != null){
                searcher(noder.left, current);
            }
            if(noder.right != null){
                searcher(noder.right, current);
            }
        }
        current = current.substring(0, current.length() - 1);
    }
}